lib/OpenQA/Git/MaintenanceOutage.pm

```diff
@@ -0,0 +1,33 @@
+# Copyright SUSE LLC
+# SPDX-License-Identifier: GPL-2.0-or-later
+
+package OpenQA::Git::MaintenanceOutage;
+use Mojo::Base -strict, -signatures;
+use Mojo::File 'path';
+use Time::Seconds qw(ONE_HOUR);
+
+# Remove or rename these "constants" in favor of a dynamic path
+use constant MAX_DURATION_SECONDS => ONE_HOUR / 2;
+
+use constant {
+    SKIP => 'SKIP',
+    FAIL => 'FAIL',
+};
+
+sub _state_file ($app) {
+    path($ENV{OPENQA_GIT_SERVER_MAINTENANCE_FILE} // ($app->config->{global}->{cache_dir} // '/var/lib/openqa/cache') . '/gitlab_maintenance.lock');
+}
+
+sub remove_state_file ($app) { _state_file($app)->remove }
+
+sub decide_outcome ($app) {
```

Comment:
I find the subroutine name a bit generic. Most subroutines are kind of deciding on an outcome :)
Maybe `check_availability` or something?